lib/saml/user_attributes/ssoe.rb

```diff
@@ -323,41 +321,23 @@ def mhv_iens
         mhv_iens.append(safe_attr('va_eauth_mhvuuid')).reject(&:nil?).uniq
       end
 
-      def mhv_inbound_outbound
+      def mhv_outbound_redirect
         tracker = SAMLRequestTracker.find(@tracker_uuid)
-        tracker&.payload_attr(:skip_dupe) == 'mhv'
-      end
-
-      def check_mhv_icn_mismatch
-        icn_val = safe_attr('va_eauth_icn')
-        mhvicn_val = safe_attr('va_eauth_mhvicn')
-        if icn_val.present? && mhvicn_val.present? && icn_val != mhvicn_val
-          raise SAML::UserAttributeError, SAML::UserAttributeError::ERRORS[:mhv_icn_mismatch]
-        end
+        tracker&.payload_attr(:skip_dupe_id_validation) == 'true'
       end
 
       def birls_id_mismatch?
         attribute_has_multiple_values?('va_eauth_birlsfilenumber')
       end
 
-      def check_corp_id_mismatch
-        check_id_mismatch(mvi_ids[:vba_corp_ids], :multiple_corp_ids)
-      end
-
-      def check_id_mismatch(ids, multiple_ids_error_type, mhv_inbound_outbound: false)
+      def check_id_mismatch(ids, multiple_ids_error_type, mhv_outbound_redirect: false)
```

Comment:
`mhv_outbound_redirect` doesn't need to be an argument. It's already a method in the class so you have access to it.

Reply:
Pretty sure we need it as an argument so that we can force it to `false` in the case of the EDIPI check, even for mhv_outbound_redirect we still want the normal behavior: throwing an error & preventing login. Or am I misinterpreting what you're saying?